### ISSUE TITLE
Styles are not enqueued with explicit dependencies

### ISSUE THREAD

**Issue by slavicd** [woocommerce/storefront#1299]
## Overview

If a child theme enqueues a stylesheet with a defined Storefront dependency, like so:
```php
wp_enqueue_style(
    'child-theme-style',
    get_stylesheet_directory_uri() .'/assets/styles/main.css',
    ['storefront-woocommerce-style']
);
```
it will break the order of the style tags of Storefront itself, which can (will) lead to broken styling _(in this particular example, `storefront-woocommerce` stylesheet will get enqueued before `storefront-icons` which will break the breadcrumbs.)_.

##  To Reproduce

1. Start with a clean Storefront
2. Add and activate a rudimentary child theme (style and functions.php)
3. Enqueue a child style in a manner similar to the above code (declare a dependency on `storefront-woocommerce-style`)
4. Observe the bug with the breadcrumbs

#### Example of broken breadcrumbs
![image](https://user-images.githubusercontent.com/1071705/77229308-d79e5b00-6b95-11ea-8713-72924086dd40.png)

#### Workaround

This could be mitigated by supplying a priority higher than `30` to `wp_enqueue_scripts` (for this particular issue), but is, arguably, not the most easily maintainable way to go about this.

#### Fix
Explicitly declare dependencies of styles in Storefront, rather than rely on the implicit order of enqueue, so that WP makes the right choice when enqueueing, even with dependencies coming from "outside".

**Comment by haszari** [comment]
Thanks for reaching out @slavicd. Can you provide a bit more detail about the specifics of how this issue affects you? The [issue template](https://github.com/woocommerce/storefront/blob/master/.github/ISSUE_TEMPLATE/1-Bug-report.md) is a good starting point, please provide as much detail and context as you can.

The docs for how to enqueue styles in a theme [are here](https://developer.wordpress.org/themes/basics/including-css-javascript/) - if Storefront is not following these guidelines please provide details so this can be addressed.

**Comment by slavicd** [comment]
I do feel like there is sufficient details as to what the issue is @haszari

**Comment by haszari** [comment]
Thanks for clarifying @slavicd. I had a quick look to see if / why `storefront-woocommerce-style` depends on `storefront-icons` or `storefront-style`. 

As far as I can tell, there's one use of FontAwesome icons in `storefront-woocommerce-style`, for password inputs: https://github.com/woocommerce/storefront/blob/master/assets/css/woocommerce/woocommerce.scss#L1606

We'll need to do some investigation to ensure we get the dependencies right for each stylesheet. Looks like you've found one potential missing/implicit dependency - feel free to add comments with more info. Or if you prefer, open a PR with a suggested fix.

I'm a little surprised that this issue hasn't been reported before. You might also want to take a look at Storefront child themes [Deli](https://github.com/woocommerce/deli) or [Boutique](https://github.com/woocommerce/boutique) to see how they handle this (or if they have the same issue you're experiencing).

**Comment by Aljullu** [comment]
> I'm a little surprised that this issue hasn't been reported before. You might also want to take a look at Storefront child themes Deli or Boutique to see how they handle this (or if they have the same issue you're experiencing).

It looks like child themes have been relying on the the work-around exposed by @slavicd to enqueue styles:

https://github.com/woocommerce/deli/blob/master/inc/class-deli.php#L25
https://github.com/woocommerce/boutique/blob/master/inc/class-boutique.php#L24

I also checked [Bistro](https://woocommerce.com/products/bistro/) and it does the same.

I agree it would be nice to fix this to make it easier to write child themes without the need to rely in this kind of hacks.

**Comment by haszari** [comment]
Thanks for reporting @slavicd and thanks for investigating @Aljullu - I've added this to the next milestone, as it seems there's an issue here we can fix.

@Aljullu  - you seem to have a good handle on the issue here, could you add a comment with next steps to clarify the problem and the intended fix/improvement? Thank you :)

**Comment by haszari** [comment]
Hi there @slavicd  - this has been fixed in Storefront 2.5.8-rc.1. If you like, you can [download the zip from here](https://github.com/woocommerce/storefront/releases/tag/version%2F2.5.8-rc.1) and test the fix. Thank you :)
